### PR TITLE
Code Coverage

### PR DESCRIPTION
**Pull request by Dhruv-J**
This PR defines the thresholds at which code coverage is run, and outlines behavior for the comment bot to follow.

### CODE REVIEW

**Comment by elton-furtado** [comment]
Can you configure all the required flags similar to antrea in codecov.yml and kind.yml

**Comment by Dhruv-J** [comment]
> Can you configure all the required flags similar to antrea in codecov.yml and kind.yml

Yeah, I thought the existing code in this PR should do that, as I've replicated the setup in go.yml for kind.yml, and updated codecov.yml identically for both. However, only the unit-tests flag, which is in go.yml, is working correctly. I'm trying to look into what could be causing it and if there's additional setup I'm missing. It may be worth it to release flag support for just unit-tests initially, and update the flags to include kind-e2e-tests when we reach 60% coverage. This way, any PRs made during the course of the 0.6 release can still be informational and contribute to reaching a higher coverage percentage. The flags will only provide a more visual breakdown, but will not change the actual coverage percentage.

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+flag_management:
+  default_rules:
+    carryforward: true
+ignore:
+  - "**/testing/*.go"
+  - "**/mock_*.go"
+  - "**/*generate*.go"
+  - "pkg/client"
+  - "**/pkg/client"
+  - "**/*.pb.go"
```
Comment (tushartathgur) [inline codecov.yml:44]: Don’t need this check yet

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - "third_party"
```
Comment (tushartathgur) [inline codecov.yml:45]: Add line to the end

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        flags:
+          - unit-tests
+
+flag_management:
+  default_rules:
+    carryforward: true
+ignore:
+  - "**/testing/*.go"
+  - "**/mock_*.go"
+  - "**/*generate*.go"
```
Comment (tushartathgur) [inline codecov.yml:41]: Don’t need to check for pkg/api as well, no tests are written for the folder yet.

**Review by tushartathgur — COMMENTED** [review]
initial comments for ignoring files

File: codecov.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        target: auto
+        threshold: 5%
+        base: auto
+        only_pulls: true
+        flags:
+          - unit-tests
+    project:
+      default:
+        target: auto
+      theia-unit-tests:
```
Comment (dreamtalen) [inline codecov.yml:30]: `e2e-tests` and `kind-e2e-tests` are missing here

**Review by dreamtalen — APPROVED** [review]
This code coverage yml does not include e2e-tests and kind-e2e-tests flags yet, so please update the commit message, PR description, and PR title to reflect that only unit-tests are supported for now. 
It would also be good to create a separate issue to track the addition of those two missing flags. 
Everything else looks good to me.

**Pull request merged**